staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -312,6 +312,11 @@ func (f *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID strin
 // InstanceShutdownByProviderID returns true if the instances is in safe state to detach volumes
 func (f *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
 	f.addCall("instance-shutdown-by-provider-id")
+
+	if providerID == "" {
+		return false, fmt.Errorf("cannot shutdown instance with empty providerID")
```

Comment:
Is it possible to add a test case where calling `instances.InstanceID(ctx, types.NodeName(node.Name))` returns no providerID? In this case we would return early and not add the taint

Reply:
> I may be missing something but if I read correctly through MonitorNodes it seems to me that node with no providerID would be deleted as it hits the path with

I was referring to InstanceID returning no providerID but not returning the `InstanceNotFound` error. I think we only delete if InstanceID returns InstanceNotFound